src/test/java/org/fogbowcloud/manager/core/AaControllerTest.java

```diff
@@ -48,112 +48,174 @@ public void setUp() {
     	
     }
     
+    //test case: Check if authenticate method throws no exception when the federation token is valid.
     @Test
     public void testAuthenticate() throws UnauthenticatedUserException {
+    	//set up
     	Mockito.when(this.federationIdentityPluginMock.isValid(Mockito.anyString())).thenReturn(true);
+    	
+    	//exercise/verify
     	this.aaController.authenticate(Mockito.anyString());
     }
     
+    //test case: Check if authenticate method throws Unauthenticated exception when the federation token is valid. 
     @Test (expected = UnauthenticatedUserException.class)
     public void testAuthenticateWhenUnauthenticatedUserException() throws UnauthenticatedUserException {
+    	//set up
     	Mockito.when(this.federationIdentityPluginMock.isValid(Mockito.anyString())).thenReturn(false);
+    	
+    	//exercise/verify
     	this.aaController.authenticate(Mockito.anyString());
     }
     
+    //test case: Check if authenticate method throws no exception when the operation is valid.
```

Comment:
typo. authorize instead of authenticate in line 71, right?

Reply:
below method also have typos

Reply:
copy-paste trouble